### ISSUE TITLE
Pass something besided string to `getOptionLabel()`?

### ISSUE THREAD

**Issue by jkjustjoshing** [JedWatson/react-select#4134]
The [docs](https://react-select.com/props) for `getOptionLabel` indicate that the only valid return type is a `string`.
<img width="574" alt="image" src="https://user-images.githubusercontent.com/813192/88094395-7a291380-cb61-11ea-9269-66cd0b2effef.png">

However, it seems like a return value of any React node will work, such as the following:
```jsx
getOptionLabel={(data) => <b>{data.name}</b>}
```

Should/can the docs be updated to show that anything React can render is an acceptable return type?

**Comment by jkjustjoshing** [comment]
Also, if this is accepted, the [DefinitelyTyped type](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/react-select/src/builtins.d.ts#L7) should be updated

**Comment by jkjustjoshing** [comment]
I just realized that this breaks the `aria-live` region / screen reader support, since the return value for `getOptionLabel` is concatenated with a bunch of other strings. When that's a JSX object you get the screen reader reading `[object Object]` which obviously isn't good. Closing ticket.

**Comment by jkjustjoshing** [comment]
Actually, would it make sense to log a console warning if a non-string is passed? Most people probably aren't verifying that their screen readers still work, and might do what I did: try it out with some JSX, realize it works, and leave it there.

**Comment by ebonow** [comment]
@jkjustjoshing Thanks for bringing this up.

It's likely intended that the types in the documentation should be updated to return a Node. I will add a note to follow through on this.

Regarding aria-live regions, I also agree this can break screen-readers. I'd like to propose a solution and see if this sounds good to you.
1. If JSX is passed in, parent node should contain an aria-label.
2. If JSX is passed in, aria-live config should use this value.

Thoughts?

**Comment by ebonow** [comment]
@jkjustjoshing Thank you for bringing this up. I am look at accessibility within react-select and these functions are critical to it working as intended. I have linked to this issue in #4407 with the hopes of getting more attention around the desired implementation and enforcement of best practices.

**Comment by ebonow** [comment]
@jkjustjoshing Thank you for bringing this up. I am currently looking at accessibility within react-select and these functions are critical to it working as intended. I have linked to this issue in #4407 with the hopes of getting more attention around the desired implementation and enforcement of best practices.
